pkg/config/config.go

```diff
@@ -839,23 +839,24 @@ func InitConfig(config Config) {
-	config.BindEnvAndSetDefault("external_metrics_provider.refresh_period", 30)           // value in seconds. Frequency of calls to Datadog to refresh metric values
-	config.BindEnvAndSetDefault("external_metrics_provider.batch_window", 10)             // value in seconds. Batch the events from the Autoscalers informer to push updates to the ConfigMap (GlobalStore)
-	config.BindEnvAndSetDefault("external_metrics_provider.max_age", 120)                 // value in seconds. 4 cycles from the Autoscaler controller (up to Kubernetes 1.11) is enough to consider a metric stale
-	config.BindEnvAndSetDefault("external_metrics.aggregator", "avg")                     // aggregator used for the external metrics. Choose from [avg,sum,max,min]
-	config.BindEnvAndSetDefault("external_metrics_provider.bucket_size", 60*5)            // Window to query to get the metric from Datadog.
-	config.BindEnvAndSetDefault("external_metrics_provider.rollup", 30)                   // Bucket size to circumvent time aggregation side effects.
-	config.BindEnvAndSetDefault("external_metrics_provider.wpa_controller", false)        // Activates the controller for Watermark Pod Autoscalers.
-	config.BindEnvAndSetDefault("external_metrics_provider.use_datadogmetric_crd", false) // Use DatadogMetric CRD with custom Datadog Queries instead of ConfigMap
-	config.BindEnvAndSetDefault("kubernetes_event_collection_timeout", 100)               // timeout between two successful event collections in milliseconds.
-	config.BindEnvAndSetDefault("kubernetes_informers_resync_period", 60*5)               // value in seconds. Default to 5 minutes
-	config.BindEnvAndSetDefault("external_metrics_provider.config", map[string]string{})  // list of options that can be used to configure the external metrics server
-	config.BindEnvAndSetDefault("external_metrics_provider.local_copy_refresh_rate", 30)  // value in seconds
-	config.BindEnvAndSetDefault("external_metrics_provider.chunk_size", 35)               // Maximum number of queries to batch when querying Datadog.
+	config.BindEnvAndSetDefault("external_metrics_provider.endpoint", "")                   // Override the Datadog API endpoint to query external metrics from
+	config.BindEnvAndSetDefault("external_metrics_provider.api_key", "")                    // Override the Datadog API Key for external metrics endpoint
+	config.BindEnvAndSetDefault("external_metrics_provider.app_key", "")                    // Override the Datadog APP Key for external metrics endpoint
+	config.SetKnown("external_metrics_provider.endpoints")                                  // List of redundant endpoints to query external metrics from
+	config.BindEnvAndSetDefault("external_metrics_provider.refresh_period", 30)             // value in seconds. Frequency of calls to Datadog to refresh metric values
+	config.BindEnvAndSetDefault("external_metrics_provider.only_use_datadogmetrics", false) // Ignore non-DatadogMetric HPA external metric queries
```

Comment:
I think the parameter name is not very descriptive. It may be easier to understand something like `external_metrics_provider.enable_autogen_datadogmetrics` with a default as `true`.

Reply:
Indeed - I am not sure what the most appropriate name for this configuration option is. I thought about focusing on autogeneration as well, but this feature also prevents non-DatadogMetric queries from being included in the Datadog Metrics API query. 

For comparison, if the datadog-cluster-agent ClusterRole is modified to disallow `DatadogMetric` creates, the autogenerated DatadogMetrics objects won't be created, but the metrics query will still be sent to the Datadog API.

I can switch to `external_metrics_provider.enable_autogen_datadogmetrics` if you think that's sufficient to describe the functionality clearly. If not, I'd be willing to discuss other proposals.

Reply:
Indeed, but modifying the ClusterRole is a workaround, it does still send the metrics because the autoscaler watcher still works properly and creates the proper state in the internal store. Modifying the role prevents the controller to properly synchronize the internal store and Kubernetes objects.

With the code change you made no autogen means no object will be created in the internal store, leading to no queries being send as well, which is, I believe, the clean way to go and the expected behaviour when autogen is not activated.